src/inc/til/rectangle.h

```diff
@@ -13,9 +13,84 @@ class RectangleTests;
 
 namespace til // Terminal Implementation Library. Also: "Today I Learned"
 {
+    class _rectangle_const_iterator
+    {
+    public:
+        constexpr _rectangle_const_iterator(point topLeft, point bottomRight) :
+            _topLeft(topLeft),
+            _bottomRight(bottomRight),
+            _current(topLeft)
+        {
+        }
+
+        constexpr _rectangle_const_iterator(point topLeft, point bottomRight, point start) :
+            _topLeft(topLeft),
+            _bottomRight(bottomRight),
+            _current(start)
+        {
+        }
+
+        _rectangle_const_iterator& operator++()
+        {
+            ptrdiff_t nextX;
+            THROW_HR_IF(E_ABORT, !::base::CheckAdd(_current.x(), 1).AssignIfValid(&nextX));
+
+            if (nextX >= _bottomRight.x())
+            {
+                ptrdiff_t nextY;
+                THROW_HR_IF(E_ABORT, !::base::CheckAdd(_current.y(), 1).AssignIfValid(&nextY));
+                _current = { _topLeft.x(), nextY };

```

Comment:
just to be clear: the _end_ is the same thing Carlos defined as the end -- the single cell off the bottom left?

Reply:
Yes, I'll make it a comment. When we add the different walk directions, it'll be in different places. It's just "one more" of whatever walk algorithm it's using.